src/openrct2/ride/TrackPaint.cpp

```diff
@@ -2201,3 +2216,123 @@ void track_paint(paint_session* session, uint8_t direction, int32_t height, cons
         }
     }
 }
+
+#ifdef __ENABLE_LIGHTFX__
+void track_paint_add_kiosk_lights(CoordsXY mapPosition, uint8_t direction, int32_t height, uint8_t zOffset)
+{
+    switch (get_current_rotation())
+    {
+        case 0:
+            lightfx_add_3d_light_magic_from_drawing_tile(mapPosition, 0, 16, height + zOffset, LIGHTFX_LIGHT_TYPE_LANTERN_3);
```

Comment:
There is far too much repetition in this code we have rotation functions there should be a need to write out all the different rotations.